lib/Sema/TypeCheckType.cpp

```diff
@@ -4273,6 +4274,7 @@ class ExistentialTypeVisitor
           Ctx.Diags.diagnose(comp->getNameLoc(),
                              diag::existential_requires_any,
                              alias->getDeclaredInterfaceType(),
+                             ExistentialType::get(alias->getDeclaredInterfaceType()),
```

Comment:
Is there a more appropriate way to get the existential type for the alias (like the `getExistentialType()` on the proto case above) or is this the right way here?

Reply:
Not at the moment, but we could certainly add `getExistentialType()` to `TypeAliasDecl` if it's helpful! The implementation of `ProtocolDecl::getExistentialType` does exactly what you've written here:

```cpp
  /// Returns the existential type for this protocol.
  Type getExistentialType() const {
    return ExistentialType::get(getDeclaredInterfaceType());
  }
```